src/query/shorthand.ts

```diff
@@ -49,6 +50,59 @@ export function vlSpec(vlspec: ExtendedUnitSpec,
   return spec(specQ);
 }
 
+const CHANNEL_SUPPORTS_AXIS: Dict<boolean> =
+  [Channel.X, Channel.Y, Channel.ROW, Channel.COLUMN]
+    .reduce((m, channel) => {
+      m[channel] = true;
+      return m;
+    }, {} as Dict<boolean>);
+
+const CHANNEL_SUPPORTS_LEGEND: Dict<boolean> =
+  [Channel.COLOR, Channel.OPACITY, Channel.SIZE, Channel.SHAPE]
+    .reduce((m, channel) => {
+      m[channel] = true;
+      return m;
+    }, {} as Dict<boolean>);
+
+const CHANNEL_SUPPORTS_SCALE: Dict<boolean> =
+  [Channel.X, Channel.Y, Channel.COLOR, Channel.OPACITY, Channel.OPACITY, Channel.ROW, Channel.COLUMN, Channel.SIZE, Channel.SHAPE]
+    .reduce((m, channel) => {
+      m[channel] = true;
+      return m;
+    }, {} as Dict<boolean>);
+
+const CHANNEL_SUPPORTS_SORT: Dict<boolean> =
+  [Channel.X, Channel.Y, Channel.PATH, Channel.ORDER]
+    .reduce((m, channel) => {
+      m[channel] = true;
+      return m;
+    }, {} as Dict<boolean>);
+
+export const CHANNEL_SUPPORTS_PROPERTY =
```

Comment:
Maybe I should just give you the answer.  But isn't this obvious?  I don't know why you need the complicated 2 layers of `reduce`.   

```
PROPERTY_SUPPORTED_CHANNELS = {
   sort: {x: true, ...}, 
   axis: {x: true, ...},
   ...
}
```